yt/frontends/enzo/simulation_handling.py

```diff
@@ -656,18 +656,18 @@ def _check_for_outputs(self, potential_outputs):
                 "%s%s" % (dir_key, index),
                 "%s%s" % (output_key, index),
             )
-            if os.path.exists(filename):
-                try:
-                    ds = load(filename)
-                    if ds is not None:
-                        my_storage.result = {
-                            "filename": filename,
-                            "time": ds.current_time.in_units("s"),
-                        }
-                        if ds.cosmological_simulation:
-                            my_storage.result["redshift"] = ds.current_redshift
-                except YTOutputNotIdentified:
-                    mylog.error("Failed to load %s", filename)
+            try:
+                ds = load(filename)
+                my_storage.result = {
+                    "filename": filename,
+                    "time": ds.current_time.in_units("s"),
+                }
+                if ds.cosmological_simulation:
+                    my_storage.result["redshift"] = ds.current_redshift
+            except OSError:
```

Comment:
can we move this up so that we do a minimal catching?

Reply:
good idea. This and similar suggestions should be addressed now !
Note that I also changed it so that the logging message happens with both OSError and YTOutputNotIdentifiedError (I don't really see why I treated them separately in retrospect)